### PR TITLE
SocialIQA

### PR DESCRIPTION
**Pull request by pruksmhc**
From: https://arxiv.org/pdf/1904.09728.pdf

The RoBERTa representation is getting 0.738 accuracy, where the SOTA is 0.66.

### CODE REVIEW

**Comment by zphang** [comment]
Should it be SocialIQA?

**Comment by pruksmhc** [comment]
What do you mean? Also, this is still WIP so don't formally review yet.

**Comment by sleepinyourhat** [comment]
The name of the task is SocialIQA with an 'I'.

**Comment by pruksmhc** [comment]
Oh wow I completely did not see that second "l" - thanks for pointing that out!

File: jiant/tasks/tasks.py [hunk 1]
```diff
@@ -2700,6 +2700,106 @@ class MultipleChoiceTask(Task):
     pass
 
 
+@register_task("SociallQA", rel_path="SociallQA/")
```
Comment (W4ngatang) [inline jiant/tasks/tasks.py]: maybe I'm crazy, but are these currently spelled as "Social*l*QA" and not "Social*I*QA" (capital I vs lowercase L)

File: jiant/tasks/tasks.py [hunk 2]
```diff
@@ -2700,6 +2700,106 @@ class MultipleChoiceTask(Task):
...
+                context = example["context"]
+                choice1 = example["answerA"]
+                choice2 = example["answerB"]
+                choice3 = example["answerC"]
+                question = example["question"]
+                choice = [
+                    tokenize_and_truncate(self._tokenizer_name, choice, self.max_seq_len)
+                    for choice in [choice1, choice2, choice3]
+                ]
+                targ = example["correct"] if "correct" in example else 0
```
[the remark this answers is not in the document]
Comment (pruksmhc) [reply jiant/tasks/tasks.py]: This should only be 0 for the test data

File: jiant/tasks/tasks.py [hunk 3]
```diff
@@ -2700,6 +2700,106 @@ class MultipleChoiceTask(Task):
...
+                targs.append(targ)
+            return [contexts, choices, questions, targs]
+
+        self.train_data_text = _load_split(os.path.join(self.path, "socialIQa_v1.4_trn.jsonl"))
+        self.val_data_text = _load_split(os.path.join(self.path, "socialIQa_v1.4_dev.jsonl"))
+        self.test_data_text = _load_split(os.path.join(self.path, "socialIQa_v1.4_tst.jsonl"))
+        self.sentences = (
+            self.train_data_text[0]
+            + self.val_data_text[0]
+            + [choice for choices in self.train_data_text[1] for choice in choices]
```
[the remark this answers is not in the document]
Comment (pruksmhc) [reply jiant/tasks/tasks.py]: Yes exactly - train_data_text[1] are the various choices, and train_data_text[0] is the one context.

**Pull request merged**